Homework-1-NodeJS-Fundamentals/task2/index.js

```diff
@@ -0,0 +1,55 @@
+import csv from 'csvtojson';
+import { existsSync, mkdirSync } from 'fs';
+import { createWriteStream } from 'fs';
+import { Transform } from "stream";
+import { StringDecoder } from 'string_decoder';
+
+if (!existsSync('./results')) {
+    mkdirSync('./results');
+}
+
+const csvFilePath='./csv/source.csv';
+const writable1 = createWriteStream('./results/result1.txt');
+const writable2 = createWriteStream('./results/result2.txt');
+
+const formatOutput = new Transform({
+    transform(chunk, encoding, callback) {
+        if (encoding === 'buffer') {
+            const decoder = new StringDecoder('utf-8');
+            chunk = decoder.write(chunk);
+        }
+
+        chunk = JSON.parse(chunk);
+        delete chunk['Amount'];
+
+        callback(null, `${JSON.stringify(chunk)}\n`);
+    }
+});
+
+/**
+ * @desc it's possible to ignore Amount column from parsing not to have it in output file
+ */
+csv({ignoreColumns: /Amount/})
+    .fromFile(csvFilePath)
+    .subscribe((json,lineNumber)=>{
+        return new Promise((resolve) => {
+            resolve();
+        })
+    })
+    .pipe(writable1)
+    .on('error', err => console.log(`Something went wrong! Error: ${err.message}`))
+
+/**
+ * @desc transform stream is used to get rid of 'Amount' property in output file
+ */
+csv()
```

Comment:
Can you maybe add an example with usage of `pipeline`? It won't be much different but it's a part of requirements.

Reply:
Added example with `pipeline`